test/cljam/algo/vcf_indexer_test.clj

```diff
@@ -0,0 +1,138 @@
+
+    (do (vcf-indexer/create-index test-vcf-complex-gz-file
+                                  tmp-csi-file {})
+        (let [csi ^CSI (csi/read-index tmp-csi-file)]
+          (is (= (.n-ref csi) 4))
+          (is (= (.min-shift csi) 14))
+          (is (= (.depth csi) 5))
+          (is (= (.loffset csi)
+                 {0 {1 3904},
+                  1 {32769 3973},
+                  2 {114704385 4031},
+                  3 {1 4783}}))
+          (is (= (chunks->maps (.bidx csi))
+                 {0 {4681 [{:beg 3904, :end 3973}]},
+                  1 {4683 [{:beg 3973, :end 4031}]},
+                  2 {11682 [{:beg 4031, :end 4783}]},
+                  3 {4681 [{:beg 4783, :end 106168320}]}}))))
+
+    (do (vcf-indexer/create-index test-vcf-complex-gz-file
+                                  tmp-csi-file {:shift 18 :depth 5})
+        (let [csi ^CSI (csi/read-index tmp-csi-file)]
+          (is (= (.n-ref csi) 4))
+          (is (= (.min-shift csi) 18))
+          (is (= (.depth csi) 5))
+          (is (= (.loffset csi)
+                 {0 {1 3904},
+                  1 {1 3973},
+                  2 {114556929 4031},
+                  3 {1 4783}}))
+          (is (= (chunks->maps (.bidx csi))
+                 {0 {4681 [{:beg 3904, :end 3973}]},
+                  1 {4681 [{:beg 3973, :end 4031}]},
+                  2 {5118 [{:beg 4031, :end 4783}]},
+                  3 {4681 [{:beg 4783, :end 106168320}]}}))))))
+
+(deftest about-vcf-changed-chr-order
+  (with-before-after {:before (prepare-cache!)
+                      :after (clean-cache!)}
+    (do (vcf-indexer/create-index test-vcf-changed-chr-order-file
+                                  tmp-csi-file {:shift 18 :depth 5})
+        (let [csi ^CSI (csi/read-index tmp-csi-file)]
+          (is (= (chunks->maps (.bidx csi))
+                 {0 {4681 [{:beg 2096, :end 2205}]},
+                  1 {4681 [{:beg 1647, :end 1811}],
+                     4685 [{:beg 1811, :end 2096}]},
+                  2 {4681 [{:beg 2205, :end 70975488}]}}))
+
+          (is (= (.loffset csi)
+                 {0 {1 2096},
+                  1 {1 1647, 1048577 1811},
+                  2 {1 2205}}))))))
+
+(deftest about-vcf-changed-chr-order-field-less
+  (with-before-after {:before (prepare-cache!)
+                      :after (clean-cache!)}
+    (do (vcf-indexer/create-index test-vcf-changed-chr-order-field-less-file
+                                  tmp-csi-file {:shift 18 :depth 5})
+        (let [csi ^CSI (csi/read-index tmp-csi-file)]
+          (is (= (chunks->maps (.bidx csi))
+                 {0 {4681 [{:beg 1597, :end 1761}],
+                     4685 [{:beg 1761, :end 2046}]},
+                  1 {4681 [{:beg 2046, :end 2155}]},
+                  2 {4681 [{:beg 2155, :end 70057984}]}}))
+
+          (is (= (.loffset csi)
+                 {0 {1 1597, 1048577 1761},
+                  1 {1 2046},
+                  2 {1 2155}}))))))
+
```

Comment:
Would you add these test cases here

```clojure
(deftest csi-comparison-test
  (let [csi ^CSI (csi/read-index test-vcf-various-bins-csi-file)
        computed (with-open [r (vcf/reader test-vcf-various-bins-gz-file)]
                   (csi/offsets->index (vcf/read-file-offsets r) 14 6))]
    (is (= (.n-ref csi) (.n-ref ^CSI computed)))
    (is (= (.min-shift csi) (.min-shift ^CSI computed)))
    (is (= (.depth csi) (.depth ^CSI computed)))
    (is (= (.bidx csi) (.bidx ^CSI computed)))
    (is (= (.loffset csi) (.loffset ^CSI computed)))))

(deftest-remote large-csi-comparison-test
  (with-before-after {:before (prepare-cavia!)}
    (let [csi ^CSI (csi/read-index test-large-vcf-csi-file )
          computed (with-open [r (vcf/reader test-large-vcf-file)]
                     (csi/offsets->index (vcf/read-file-offsets r) 14 6))]
      (is (= (.n-ref csi) (.n-ref ^CSI computed)))
      (is (= (.min-shift csi) (.min-shift ^CSI computed)))
      (is (= (.depth csi) (.depth ^CSI computed)))
      (is (= (.bidx csi) (.bidx ^CSI computed)))
      (is (= (.loffset csi) (.loffset ^CSI computed))))))
```


and this one in `cljam.io.csi-test`?

```clojure
(deftest-remote large-read-write-test
  (with-before-after {:before (do (prepare-cavia!)
                                  (prepare-cache!)),
                      :after (clean-cache!)}
    (let [temp-csi-file (cio/file temp-dir "test_temp.csi")
          r ^CSI (csi/read-index test-large-vcf-csi-file)
          _ (csi/write-index temp-csi-file r)
          w ^CSI (csi/read-index temp-csi-file)]
      (is (= (.n-ref r) (.n-ref w)))
      (is (= (.min-shift r) (.min-shift w)))
      (is (= (.depth r) (.depth w)))
      (is (= (.bidx r) (.bidx w)))
      (is (= (.loffset r) (.loffset w))))))
```